src/blockdata/block.rs

```diff
@@ -212,7 +212,14 @@ impl Block {
         false
     }
 
+    /// Compute the transaction merkle root.
+    pub fn compute_merkle_root(&self) -> Option<TxMerkleNode> {
+        let hashes = self.txdata.iter().map(|obj| obj.txid().as_hash());
+        bitcoin_merkle_root(hashes).map(|h| h.into())
+    }
+
     /// Calculate the transaction merkle root.
+    #[deprecated(since = "0.27.1", note = "Please use `block::compute_merkle_root` instead.")]
```

Comment:
We add deprecated since the upcoming minor release, so this should be
```
#[deprecated(since = "0.28.0", note = "Please use `block::compute_merkle_root` instead.")]
```